### PR TITLE
Implemented httpOnly cookie option and a signout endpoint

### LINKED ISSUE #846: Implement HTTPOnly cookies
As mentioned by @jfresco at #836.

The disscussion here is that to do this we should change the logout functionality to make it do a request to the server, because right now we only unset the 'token' cookie to logout.

_**More info:** http://blog.codinghorror.com/protecting-your-cookies-httponly/_

### PR DESCRIPTION
**Pull request by jfresco**
This prevents programmatic access to the session cookie from the browser.
Closes #846.

### CODE REVIEW

File: lib/citizen/model.js [hunk 1]
```diff
@@ -108,7 +107,8 @@ Citizen.prototype.logged = function() {
  * @api public
  */
 
-Citizen.prototype.unload = function() {
+Citizen.prototype.unload = function(cb) {
+  var self = this;
   this.cleanup();
   this.$_path = null;
   var domain = null;
```
Comment (jfresco) [inline lib/citizen/model.js]: This variable and all its references can be safely deleted.

File: lib/signin-api/index.js [hunk 2]
```diff
@@ -58,3 +58,8 @@ app.post('/', function(req, res, next) {
     });
   }
 });
+
+app.del('/', function(req, res, next) {
+  log('sending response with cookie deletion')
```
Comment (jfresco) [inline lib/signin-api/index.js]: Not necessary.

**Comment by gvilarino** [comment]
@jfresco shouldn't we disable the `httpOnly` option when under `development`? This would make it easier for forkers to debug stuff. Just sayin'

**Comment by mjlescano** [comment]
@gvilarino Doing that may make work something on dev but not on prod.

**Comment by jfresco** [comment]
@gvilarino You can still see the cookies with Chrome Developer Tools or similar. In what cases would you want to get or set the cookie with client-side JavaScript for debugging purposes?

**Comment by gvilarino** [comment]
Ok, disregard my comment then. I'll trust your criterion.

**Pull request merged**